Test_Cases/AlicePrimeStringTest.java

```diff
@@ -0,0 +1,63 @@
+package com.core.programming;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * The Class AlicePrimeStringTest.
+ */
+public class AlicePrimeStringTest {
+
+	/** The test prime string. */
+	AlicePrimeString testAlicePrimeString;
+
+	/**
+	 * Setup - Instantiate AlicePrimeString
+	 */
+	@Before
+	public void setUp() {
+		testAlicePrimeString = new AlicePrimeString();
+				
+	}
+
+	@After
+	public void tearDown() {
+		testAlicePrimeString = null;
+		
+	}
+
+	/**
+	 * Test CheckPrimeString positive.
+	 */
+	@Test
+	public void testCheckPrimeStringPositive() {
+		Assert.assertEquals("Yes", testAlicePrimeString.checkPrimeString("ababb"));
+		Assert.assertEquals("Yes", testAlicePrimeString.checkPrimeString("abcccab"));
+		
+		Assert.assertEquals("No", testAlicePrimeString.checkPrimeString("abcab"));
+		Assert.assertEquals("No", testAlicePrimeString.checkPrimeString("abcdab"));
+		Assert.assertEquals("No", testAlicePrimeString.checkPrimeString("a"));
+		Assert.assertEquals("No", testAlicePrimeString.checkPrimeString("ab"));
+
+	}
+
+	/**
+	 * Test CheckPrimeString negative.
+	 */
+	@Test
+	public void testCheckPrimeStringNegative() {
+		Assert.assertEquals("Error", testAlicePrimeString.checkPrimeString(""));
+		Assert.assertEquals("Error", testAlicePrimeString.checkPrimeString(" "));
+		Assert.assertEquals("Error", testAlicePrimeString.checkPrimeString(null));
+	}
+
+	/**
+	 * Test CheckPrimeString verify logs.
+	 */
+	@Test
+	public void testCheckPrimeStringVerifyLogs() {
```

Comment:
@neerajkujha This test case is similar to positive test case, it's asserting expected value(String) with actual value. Just that this test case is meant for verifying logs manually by the developer. If we need to verify logs programmatically/using JUnit, one way of doing it is, asserting the count of `log` statements of `Level.INFO` & `Level.SEVERE`

Reply:
hmm, but logs can also be verified with the test cases you have written for positive and negative scenario. So what is the utility of just one positive test case in test logs. or do the above test cases not print the logs?

Reply:
Yes, even positive & negative test case do print logs. I believed that it would be easier to verify the logs with a dedicated test case for this purpose, since it invokes actual method to test only once. In case of positive & negative test case it has multiple invocations of the method, & it would be a little cumbersome to verify the logs. Yes I agree this test case utility isn't that significant.